Packs/McAfee-TIE/Integrations/McAfee-TIE/McAfee-TIE.py

```diff
@@ -1,38 +1,189 @@
 import demistomock as demisto
+import tempfile
 from CommonServerPython import *
 from dxlclient.client_config import DxlClientConfig
 from dxlclient.client import DxlClient
 from dxlclient.broker import Broker
 from dxltieclient import TieClient
-from dxltieclient.constants import HashType
 from datetime import datetime
+from typing import NamedTuple
+from dxltieclient.constants import FileReputationProp, FileGtiAttrib, FileEnterpriseAttrib, AtdAttrib, EpochMixin, TrustLevel,\
+    HashType, EnterpriseAttrib, FileProvider
 
 VENDOR_NAME = 'McAfee Threat Intelligence Exchange'
+OUTPUT_PREFIX = 'McAfee.TIE'
+LOWEST_TRUST_LEVEL_KEY = 'lowest_trust_level_key'
+LOWEST_SCORE_KEY = 'lowest_score_key'
 
-HASH_TYPE_KEYS = {
-    'md5': HashType.MD5,
-    'sha1': HashType.SHA1,
-    'sha256': HashType.SHA256
-}
-
-TRUST_LEVELS = {
-    '0': 'NOT_SET',
-    '1': 'KNOWN_MALICIOUS',
-    '15': 'MOST_LIKELY_MALICIOUS',
-    '30': 'MIGHT_BE_MALICIOUS',
-    '50': 'UNKNOWN',
-    '70': 'MIGHT_BE_TRUSTED',
-    '85': 'MOST_LIKELY_TRUSTED',
-    '99': 'KNOWN_TRUSTED',
-    '100': 'KNOWN_TRUSTED_INSTALLER'
-}
-
-POVIDER = {
-    '1': 'Global Threat Intelligence (GTI)',
-    '3': 'Enterprise reputation',
-    '5': 'Advanced Threat Defense (ATD)',
-    '7': 'Web Gateway (MWG)'
-}
+
+DXLConfigFiles = NamedTuple('DXLConfigFiles', [('broker_ca_bundle', str),
+                                               ('cert_file', str),
+                                               ('private_key', str),
+                                               ('broker_urls', list[str]),
+                                               ])
+
+
+class McAfeeTieConstants(object):
+
+    PROVIDERS = {
+        FileProvider.GTI: 'Global Threat Intelligence (GTI)',
+        FileProvider.ENTERPRISE: 'Enterprise reputation',
+        FileProvider.ATD: 'Advanced Threat Defense (ATD)',
+        FileProvider.MWG: 'Web Gateway (MWG)'
+    }
+
+    TRUST_LEVELS = {
+        TrustLevel.KNOWN_TRUSTED_INSTALLER: 'KNOWN_TRUSTED_INSTALLER',
+        TrustLevel.KNOWN_TRUSTED: 'KNOWN_TRUSTED',
+        TrustLevel.MOST_LIKELY_TRUSTED: 'MOST_LIKELY_TRUSTED',
+        TrustLevel.MIGHT_BE_TRUSTED: 'MIGHT_BE_TRUSTED',
+        TrustLevel.UNKNOWN: 'UNKNOWN',
+        TrustLevel.MIGHT_BE_MALICIOUS: 'MIGHT_BE_MALICIOUS',
+        TrustLevel.MOST_LIKELY_MALICIOUS: 'MOST_LIKELY_MALICIOUS',
+        TrustLevel.KNOWN_MALICIOUS: 'KNOWN_MALICIOUS',
+        TrustLevel.NOT_SET: 'NOT_SET',
+    }
+
+    HASH_TYPE_KEYS = {
+        'md5': HashType.MD5,
+        'sha1': HashType.SHA1,
+        'sha256': HashType.SHA256
+    }
+
+
+class GeneralFileReputationParser(object):
```

Comment:
```suggestion
class GeneralFileReputationParser(abc.ABC):
```

you're not in ~kansas~ java anymore